### PR TITLE
Network prune filters for http api (compat and libpod)

### PR DESCRIPTION
**Pull request by jmguzik**
/kind feature
Prune filters are missing not implemented in HTTP API. This PR tackles the topic in compat and libpod API.

### CODE REVIEW

File: libpod/network/netconflist.go
Comment (jmguzik) [inline libpod/network/netconflist.go:332]: @containers/podman-maintainers Windows and OS X cross builds are failing because of syscall.Stat_t. As far as I know, there is no way to get timestamp created from data structures as in the case of the containers. What would be your recommended way to pass cross builds? Making arch-specific code here?

**Comment by jmguzik** [comment]
It seems there is timezone issue (maybe due to time change in USA):
```
           Expected
               <string>: '04 EDT'
           to contain substring
               <string>: EST
         
```

File: libpod/network/netconflist.go
Comment (Luap99) [reply libpod/network/netconflist.go:332]: I will open a PR to fix this. We should not use `libpod/network` client side.

Comment (jmguzik) [reply libpod/network/netconflist.go:332]: @Luap99 Ok, thanks. Let me know and then I will do a rebase.

Comment (Luap99) [reply libpod/network/netconflist.go:332]: You can try with #9716

**Comment by jmguzik** [comment]
sys remote ubuntu-2004 root host fails because of some podman build issues. I think it is unrelated. I would be grateful if somebody with rights would re-run.

File: libpod/network/netconflist.go
Comment (jmguzik) [reply libpod/network/netconflist.go:333]: done

File: libpod/network/netconflist.go
Comment (Luap99) [inline libpod/network/netconflist.go:260]: Move this function to `pkg/util`

File: libpod/network/netconflist.go
Comment (Luap99) [inline libpod/network/netconflist.go:282]: I think it would be good to also add the until filter for network ls

**Review by Luap99 — COMMENTED** [review]
Make sure to update the api docs in `pkg/api/server/register_networks.go`.
Also squash your commits please.

File: libpod/network/netconflist.go
Comment (jmguzik) [reply libpod/network/netconflist.go:260]: @Luap99 please be certain about this location. In the coming days I will add a unification PR which will remove code duplication. Maybe it is worth to create a new dir that could be shared between libpod and compat endpoints? Please comment on this.

File: libpod/network/netconflist.go
Comment (jmguzik) [reply libpod/network/netconflist.go:282]: It is api extension then. Docker http does not allow this:
https://docs.docker.com/engine/api/v1.41/#operation/NetworkList

Comment (jmguzik) [reply libpod/network/netconflist.go:282]: Let me know if you really want it.

File: libpod/network/netconflist.go
Comment (Luap99) [reply libpod/network/netconflist.go:260]: I thought about moving it to this file pkg/util/utils.go. I do not want `libpod/network` to be imported by non network related packages like you do in pkg/domain/filters/containers.go.

Removing code duplication between compat and libpod would definitely be great.

Comment (jmguzik) [reply libpod/network/netconflist.go:260]: pkg/util/utils.go? or pkg/util/filter_helpers.go? There will be quite a few functions there coming (depending on my free time of course to do this).

File: libpod/network/netconflist.go
Comment (Luap99) [reply libpod/network/netconflist.go:282]: I don't thinks its needed when docker doesn't have it and no one asked for it.

File: libpod/network/netconflist.go
Comment (Luap99) [reply libpod/network/netconflist.go:260]: If you want to add more a new file would is good.

Comment (jmguzik) [reply libpod/network/netconflist.go]: moved to pkg/util/filters.go this file will grow once this one gets merged.

**Comment by jmguzik** [comment]
> Make sure to update the api docs in `pkg/api/server/register_networks.go`.
> Also squash your commits please.

Commits squashed and API docs updated.

**Review by Luap99 — APPROVED** [review]
LGTM

**Comment by Luap99** [comment]
@containers/podman-maintainers PTAL

**Pull request merged**
